packages/react-dev-utils/WebpackDevServerUtils.js

```diff
@@ -264,13 +266,11 @@ function prepareProxy(proxy) {
     process.exit(1);
   }
 
-  // Otherwise, if proxy is specified, we will let it handle any request.
-  // There are a few exceptions which we won't send to the proxy:
-  // - /index.html (served as HTML5 history API fallback)
-  // - /*.hot-update.json (WebpackDevServer uses this too for hot reloading)
-  // - /sockjs-node/* (WebpackDevServer uses this for hot reloading)
-  // Tip: use https://jex.im/regulex/ to visualize the regex
-  const mayProxy = /^(?!\/(index\.html$|.*\.hot-update\.json$|sockjs-node\/)).*$/;
+  // Otherwise, if proxy is specified, we will let it handle any request except for files in the public folder.
+  function mayProxy(pathname) {
```

Comment:
This patch seems to work even without the regex, but those paths still get tested (though it doesn't seem to do anything; probably because of the `text/html` check in the request).

For explicitness and safety, I'd just readd:
```diff
diff --git a/packages/react-dev-utils/WebpackDevServerUtils.js b/packages/react-dev-utils/WebpackDevServerUtils.js
index 70b5e6a..c4f1176 100644
--- a/packages/react-dev-utils/WebpackDevServerUtils.js
+++ b/packages/react-dev-utils/WebpackDevServerUtils.js
@@ -268,6 +268,13 @@ function prepareProxy(proxy, appPublicFolder) {
 
   // Otherwise, if proxy is specified, we will let it handle any request except for files in the public folder.
   function mayProxy(pathname) {
+    // There are a few exceptions which we won't send to the proxy:
+    // - /*.hot-update.(js|json) (WebpackDevServer uses this for hot reloading)
+    // - /sockjs-node/* (WebpackDevServer uses this for hot reloading)
+    // Tip: use https://jex.im/regulex/ to visualize the regex
+    if (/^(\/(.*\.hot-update\.js(on)?$|sockjs-node\/)).*$/.test(pathname)) {
+      return false;
+    }
     const maybePublicPath = path.resolve(appPublicFolder, pathname.slice(1));
     return !fs.existsSync(maybePublicPath);
   }
```